cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -150,18 +149,32 @@ class TestUnaryArithmetic : public TestBase {
     AssertArraysApproxEqual(*expected, *actual, /*verbose=*/true, equal_options_);
   }
 
-  void SetOverflowCheck(bool value = true) { options_.check_overflow = value; }
-
   void SetNansEqual(bool value = true) {
     this->equal_options_ = equal_options_.nans_equal(value);
   }
 
-  ArithmeticOptions options_ = ArithmeticOptions();
+  Options options_ = Options();
   EqualOptions equal_options_ = EqualOptions::Defaults();
 };
 
+template <typename T, typename Options>
```

Comment:
Is there any value to this subclass?

Reply:
It is the primary template for which explicit specializations are provided for `ArithmeticOptions`, `RoundOptions`, and `MRoundOptions`. I simplified these subclasses i n latest commit.

Reply:
Right, I just don't see why you couldn't have them all inherit from the Base variant.

Reply:
Ok, I reworked the template specialization as subclasses of the Base variant. Thanks for this review!